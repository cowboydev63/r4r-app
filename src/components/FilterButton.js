import React from 'react';
import PropTypes from 'prop-types';
import { Theme } from '../theme';

const FilterButton = ({ isMobileMenuOpen, toggleMobileMenu, filtersCount }) => (
    <Theme
        element="button"
        className="results__filter-button"
        onClick={ toggleMobileMenu }
    >
        { 
            !isMobileMenuOpen 
                ? 
                    `Filter${ filtersCount ? ` (${ filtersCount })` : '' }` 
                : 
                    'Done' 
        }
    </Theme>
)

FilterButton.propTypes = {
    isMobileMenuOpen: PropTypes.bool,
    toggleMobileMenu: PropTypes.func,
    filtersCount: PropTypes.number,
}

FilterButton.defaultProps = {
    isMobileMenuOpen: false,
    toggleMobileMenu: () => {},
    filtersCount: 0,
}

export default FilterButton;