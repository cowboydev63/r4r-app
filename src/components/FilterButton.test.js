import React from 'react';
import { shallow } from 'enzyme';
import FilterButton from './FilterButton';

describe('FilterButton Component', () => {

  it('renders correctly', () => {
    const wrapper = shallow(<FilterButton filtersCount={2} />);
    expect(wrapper).toMatchSnapshot();
  })
})