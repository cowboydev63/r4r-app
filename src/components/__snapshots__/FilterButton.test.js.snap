// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton Component renders correctly 1`] = `
<Theme
  className="results__filter-button"
  element="button"
  onClick={[Function]}
>
  Filter (2)
</Theme>
`;
